### PR TITLE
Update Dockerfile for .net8.0

### PR DESCRIPTION
**Pull request by sebastienros**
libmsquic updated commands
python3 tools

### CODE REVIEW

File: docker/agent/Dockerfile [hunk 1]
```diff
@@ -44,18 +44,19 @@ RUN apt-get update \
 RUN curl -LO https://packages.microsoft.com/keys/microsoft.asc && \
     echo 2cfd20a306b2fa5e25522d78f2ef50a1f429d35fd30bd983e2ebffc2b80944fa microsoft.asc| sha256sum --check - && \
     apt-key add microsoft.asc && \
-    apt-add-repository https://packages.microsoft.com/debian/11/prod && \
+    rm microsoft.asc && \
+    echo deb https://packages.microsoft.com/debian/12/prod bookworm main >> /etc/apt/sources.list.d/microsoft.list && \
```
Comment (BrennanConroy) [inline docker/agent/Dockerfile:48]: Ah yes, the ugly hack 😄

File: docker/agent/Dockerfile [hunk 2]
```diff
@@ -44,18 +44,19 @@ RUN apt-get update \
...
     apt-get update && \
     apt-get install -y libmsquic && \
-    rm -rf /var/lib/apt/lists/* microsoft.asc
```
Comment (BrennanConroy) [inline docker/agent/Dockerfile:50]: Why?

**Review by BrennanConroy — APPROVED** [review]

File: docker/agent/Dockerfile (hunk 2 above)
Comment (BrennanConroy) [reply docker/agent/Dockerfile:50]: Nvm

File: docker/agent/Dockerfile [hunk 3]
```diff
@@ -44,18 +44,19 @@ RUN apt-get update \
 RUN curl -LO https://packages.microsoft.com/keys/microsoft.asc && \
     echo 2cfd20a306b2fa5e25522d78f2ef50a1f429d35fd30bd983e2ebffc2b80944fa microsoft.asc| sha256sum --check - && \
     apt-key add microsoft.asc && \
-    apt-add-repository https://packages.microsoft.com/debian/11/prod && \
+    rm microsoft.asc && \
```
Comment (amcasey) [inline docker/agent/Dockerfile:47]: Does it matter that this is no longer `-f`?

File: docker/agent/Dockerfile (hunk 2 above)
Comment (amcasey) [reply docker/agent/Dockerfile:50]: It moved above, but I'd also like to understand why.

**Review by amcasey — APPROVED** [review]

File: docker/agent/Dockerfile (hunk 2 above)
Comment (sebastienros) [reply docker/agent/Dockerfile:50]: I took these from https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/main/src/debian/12/helix/amd64/Dockerfile#L41-L49

I don't know if that's necessary

File: docker/agent/Dockerfile (hunk 3 above)
Comment (sebastienros) [reply docker/agent/Dockerfile:47]: Maybe the other rm requires it and not this one. I tested it on 3 machines (x64 and arm64) and it worked fine.

**Pull request merged**